Run Builder Launchers

labkey-martyp opened the pull request:
- Register mechanism for run editor in ExperimentService
- Launch new run editor from runs grid
- Launch run editor for current run from run details page
- Launch run editor from sample set contents table with selected samples as either inputs or outputs
- Launch run editor from file browser with selected files as inputs or outputs

labkey-martyp committed:
Item 6854: Runbuilder launchers

- Create new run from runs grid
- Edit existing run from run details page
- Only available when provenance module is enabled
- Only available for sample derivation runs
- Sample launcher partial implementation

labkey-martyp committed:
Item 6854: Finish up sample set content run builder launch

- Samples dropdown with input/output samples options
- Samples create run button gets the selected rows (if any selected) and sets as url parameter to run builder
- Remove samples create run action.  Use onClick script instead to get selected and navigate to run builder URL
- File webpart gets experiment service registered run editors.  If they exist  and user has insert permission then add Create Run button to webpart.
- Add editor url as bean in webpart
- Popup to select input or output files on created run
- Pass file paths relative to the current container as url params to run builder

labkey-martyp committed:
Item 6854: Updates around create run from file browser

labkey-martyp committed:
Item 6854: Update link text

labkey-martyp committed:
Update style and default selected on file browser popup

labkey-kevink commented on api/src/org/labkey/api/exp/api/ExpRunEditor.java:
```diff
@@ -0,0 +1,40 @@
+import org.labkey.api.data.Container;
+import org.labkey.api.view.ActionURL;
+
+public class ExpRunEditor
```
Can you add some documentation to this class?

labkey-martyp replied:
Added class documentation

labkey-kevink commented on api/src/org/labkey/api/exp/api/ExpRunEditor.java:
```diff
@@ -0,0 +1,40 @@
+    public ActionURL getEditUrl(Container c)
+    {
+        // new action url so parameters don't get added repeatedly to _editUrl
+        ActionURL editUrl = new ActionURL(_editUrl.getLocalURIString());
```
You can also use _editUrl.clone() to avoid re-parsing the URL.

labkey-martyp replied:
Good call. fixed

labkey-kevink commented on experiment/src/org/labkey/experiment/controllers/exp/SampleSetContentsView.java:
```diff
@@ -48,6 +51,37 @@ public static ActionButton getDeriveSamplesButton(@NotNull Container container,
         return deriveButton;
     }
 
+    private String getSelectedScript(ActionURL url, boolean isOuput)
```
You don't need to change it now, but instead of passing the selected samples on the URL, you could also pass the DataRegion selectionKey and fetch the samples on the run builder page.

labkey-martyp replied:
Ah I see, I wasn't really aware of how that selectionKey worked.  But also wasn't sure if there would be more use cases where we would want to pass in sample ids.  I'll look into the selectionKey next time though and try that out.

labkey-kevink commented on filecontent/webapp/File/panel/Browser.js:
```diff
@@ -713,6 +713,17 @@ Ext4.define('File.panel.Browser', {
                 scope: this
             });
 
+            this.actionsMap.createRun = Ext4.create('File.panel.Action', {
```
I think there is a way to only enable this action if one or more files are selected, but I'm not sure.

labkey-martyp replied:
Yes there is, I updated to that action.

labkey-kevink committed:
fix concatenation of URL parameters with experimental "no question mark" URLs

labkey-martyp committed:
Run builder code review updates

labkey-martyp committed:
Merge branch 'fb_runbuilder_launch' of https://github.com/LabKey/platform into fb_runbuilder_launch

labkey-martyp committed:
Fix file browser URL

The url in the run editor is a relative URL, need to also pass in the base URL.

labkey-martyp committed:
Merge branch 'develop' into fb_runbuilder_launch

labkey-martyp committed:
Add exp.Materials and exp.Data cycle checks to run save handler

- Prevents adding materials or datas as both inputs and outputs
- Throws exception which will rollback entire save transaction

labkey-martyp committed:
Merge branch 'develop' into fb_runbuilder_launch

labkey-martyp committed:
Merge branch 'develop' into fb_runbuilder_launch

labkey-martyp committed:
Fix typo

labkey-martyp committed:
Merge branch 'develop' into fb_runbuilder_launch

labkey-tchad committed:
Merge remote-tracking branch 'origin/develop' into fb_runbuilder_launch
